[feature] Shortcodes and anchor-link.html can access lang context

southerntofu opened the pull request:
[Discussion on the forum](https://zola.discourse.group/t/i18n-lang-in-shortcodes-anchor-link/391)

I haven't updated the documentation yet. I've included lang in shortcodes and anchor-link.html, but in separate commits so you can cherry pick. I'll wait for your feedback before making changes to docs

Commit:
Pass lang to shortcodes context

Commit:
Add tests for lang in shortcodes

Commit:
Lang is passed to anchor-link.html template

Keats reviewed (commented):
Seems ok to me, can you add the docs?

Commit:
Document passing lang to shortcodes/anchor-link.html
Add a test to make sure lang can be overriden by passing an explicit argument to the shortcode,
for usage in the markdown filter where the language context is not available.

Keats commented on docs/content/documentation/content/shortcodes.md:
```diff
@@ -134,10 +134,19 @@ If you want to have some content that looks like a shortcode but not have Zola t
+Every shortcode can access some variables, beyond what you explicitly passed as parameter. These variables are explained in the following subsections:
+
+- invocation count (`nth`)
+- current language (`lang`), unless called from the `markdown` template filter (in which case it will always be `en`)
```
always be the default language, not necessarily `en`

Commit:
Update docs for more information on shortcodes+i18n+markdown() filter
